google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -24,6 +24,28 @@
     import pandas
 except ImportError:  # pragma: NO COVER
     pandas = None
+else:
+    import numpy
+
+try:
+    from shapely.geometry.base import BaseGeometry as _BaseGeometry
+except ImportError:  # pragma: NO COVER
+    _BaseGeometry = type(None)
```

Comment:
It only relates to the `else:` block in that it's done of shapely isn't present. :)

in `bq_to_arrow_array` we need to decide what arrow type to use.  Today, before this PR, we always use string for geometry, but if we have WKB data, we need to use binary.  We use WKB if:

- We have a GeoSeries, in which case we know the values are shapely.
- We look at the first non-null value and it's a shapely object, in which case it's an instance of `_BaseGeometry`.
- We look at the first non-null value and see it's `bytes`. If it's `bytes`, it can't be WKT, so we assume WKB and hope for the best.

We could use WKT with shapely objects, but WKB is more efficient.

Reply:
I added more comments to try to clarify.